content/docs/for-buildpack-authors/how-to/write-buildpacks/add-labels.md

```diff
@@ -5,6 +5,56 @@ weight=99
 
 <!--more-->
 
-This page is a stub! The CNB project is applying to [Google Season of Docs](https://developers.google.com/season-of-docs/docs/timeline) to receive support for improving our documentation. Please check back soon.
+Labels are key-value pairs, stored as strings, that are attached to an image (i.e., arbitrary metadata). Labels are used to add helpful descriptions or attributes to an application image, which are meaningful to users.
 
-If you are familiar with this content and would like to make a contribution, please feel free to open a PR :)
+Labels are usually added at the time an image is created and can also be modified at a later time. Images can have multiple labels; however, each key must be unique.
+
+## Key Points
+
+A `LABEL`
+
+* MUST specify a unique key for a given image
+* MUST specify a value to be set in the image label  
+* MUST be added as an image label on the created image metadata
+* If one key is given multiple values, the last value overwrites  
+
+## Use Cases
+
+Since adding labels to application images is optional and not needed to run containers, this property is often overlooked. The following are few reasons to why labels should be widely adopted  
+
+* Documenting versioning
+* Record licensing information
+* Including information about a project maintainer
+* Including a description of the image and  additional metadata related to the image
+* Labels can also be used to organize images  
+
+## Implementation Steps
+
+* General syntax of `LABEL` instruction is as follows:
+
+    `LABEL <key-string>=<value-string>`
+
+    `LABEL version="1.0"`
+
+### Examples  
+
+1. A `run.Dockerfile` SHOULD set the label `io.buildpacks.rebasable` to `true` to indicate that any new run image layers are safe to rebase on top of new runtime base images
+
+    * For the final image to be rebasable, all applied Dockerfiles must set this label to `true`
+
+2. A buildpack ID, buildpack version, and at least one stack MUST be provided in the OCI image config as a Label.
+
+    Label: `io.buildpacks.buildpackage.metadata`
+
+```json
+{
+  "id": "<entrypoint buildpack ID>",
+  "version": "<entrypoint buildpack version>",
+  "stacks": [
+    {
+      "id": "<stack ID>",
+      "mixins": ["<mixin name>"]
+    }
+  ]
+}
+```
```

Comment:
This example is important.  But I don't think it's the right example at this point.  If we stick with providing an exampl in shell, then the example could be a heredoc

```bash
cat << EOF > "${CNB_LAYERS_DIR}"/node-js/launch.toml
[[labels]]
key = "value"
EOF
```

An example in Go, would set the `Labels` field in a `libcnb.BuildResult` as returned by the `build` binary:

```go
func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
    result := libcnb.BuildResult{}

    result.Labels = append(result.Labels, libcnb.Label{Key: "key", Value: "value"})

    return result
}
```

The `libcnb` library will automatically take care of correctly writing `launch.toml` to the correct place.

Reply:
@AidanDelaney should I remove the first example I added or add all 3? thanks